app/models/visualization.rb

```diff
@@ -3,4 +3,20 @@ class Visualization < ApplicationRecord
   serialize :data, JSON
   has_and_belongs_to_many :samples
   belongs_to :user
+  validates :name, presence: true
+  validates :data, presence: true
+
+  delegate :count, to: :samples, prefix: true
+
+  # In the common case, a visualization will come from a single project.
+  def project_name
+    if samples.length == 1
```

Comment:
Again, this if branch is not necessary

Reply:
because to_sentence will return the same result on an array of length 1? I wasn't sure so I thought this was safer

Reply:
yes, since they are both array `to_sentence` handles it.